pkg/steps/pod.go

```diff
@@ -240,7 +240,7 @@ func (s *podStep) generatePodForStep(image string, containerResources coreapi.Re
 	pod.Spec.ServiceAccountName = s.config.ServiceAccountName
 	container := &pod.Spec.Containers[0]
 	for _, secret := range s.config.Secrets {
```

Comment:
Is it safe to assume that the secret name here will be a valid kube secret name? Otherwise this might break a lot of things

Reply:
ack. Will add testcase. Naming works for our dataset, but I'll tighten the logic to work with any valid k8s.
/hold